src/save_state.c

```diff
@@ -0,0 +1,58 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "struct_decls/struct_save_state_decl.h"
+#include "savedata.h"
+
+#include "savedata.h"
```

Comment:
**suggestion:** Since this module deals solely with save information, I think that it should be either shoved into the `savedata` folder (if we even want to keep that folder, since it's not an overlay) or keep this `save_` prefix on its name. In either case, I don't like calling this `save_state`, nor the encased `struct` `SaveState`. This is player-specific data, and the module name should reflect that.